### DISCUSSION TITLE
2nd default aes aes_* functions?

### DISCUSSION THREAD

**Discussion by EvaMaeRey** [ggplot2-extenders/ggplot-extension-club#59] [Ideas]
I really appreciate default_aes-ing especially for labels.  But then sometimes, I want a second default option.  

For pie chart, my default label is percent - it feels like the right first default, but sometimes percent and count would be nice so maybe you could do this:  aes_label_percent_count() as a wrapper for aes(label = after_stat(paste0(percent,"\n(",  count, ")"))) 

![image](https://github.com/user-attachments/assets/74a75c43-72d8-4e47-ac14-7f59d6c332c5)

For geom_circlepack label I have sizing according to area only, but if we size according to inverse nchar of the label and area we get a better result, and you could provide aes_size_area_nchar() (kind of a dumbed down text fitting) as a kind of second default.

![image](https://github.com/user-attachments/assets/9e83e891-5548-46f9-a305-3baf14ad6a78)

![image](https://github.com/user-attachments/assets/4ed17ffa-189f-434a-8c38-3d9351fd1f7f)

from ggplot2 tricks, autocontrast is kind of the same thing but I'd lean towards an aes_* prefix. 
 aes_color_after_scale_contrast_fill(), https://github.com/teunbrand/ggplot_tricks?tab=readme-ov-file#splicing-aesthetics 

Other ways?

**Comment by teunbrand** [comment]
Can't you just include the 2nd default as a computed variable and direct users to use `aes(var = after_stat(alt_value))`?

**Reply by EvaMaeRey** [reply]
yeah, I'm probably taking a more a user-side approach (similar to autocontrast) rather adding more computed vars but probably should think about adding more computed if anticipated to be useful.   I'd had in mind geom_circlepack_text_size2 or geom_pie_text_percent_count, where Stat's default_aes are modified.   But doesn't look great.  Maybe menu: geom_pie_text(text_default_option = c("percent", "count", "percent count", "count percent")

**Reply by teunbrand** [reply]
Quick question: is this *just* for formatting stuff?
Because if it is, you can have an argument that will be passed as a {glue} string.
You could have e.g. `geom_pie_text(fmt = "{count} ({percent}%)")` and internally use `glue_data()` to do the string interpolation.

**Reply by EvaMaeRey** [reply]
I'm not familiar with the fmt-glue_data move! 😮  I guess format might be the biggest area for 'second default'.

**Reply by larmarange** [reply]
> Quick question: is this *just* for formatting stuff?
> Because if it is, you can have an argument that will be passed as a {glue} string.
> You could have e.g. `geom_pie_text(fmt = "{count} ({percent}%)")` and internally use `glue_data()` to do the string interpolation.

That's a good idea
